packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -149,6 +149,21 @@ class ShaderMask extends SingleChildRenderObjectWidget {
 ///
 /// This effect is relatively expensive, especially if the filter is non-local,
 /// such as a blur.
+///
+/// The effect is applied to all existing painted content. To constrain
```

Comment:
I agree that the behavior is surprising and I probably should have filed a bug on it at the time.